ros/api/v0/hosts.py

```diff
@@ -153,6 +152,11 @@ def get(self):
             limit, offset, hosts, count
         )
 
+    # Filter systems
+    def filter_with(self, filter_name, filter_param, system_query):
+        return db.session.query(System.id).filter(System.id.in_(system_query.subquery()))\
+                .filter(System.__dict__[filter_name].ilike(f'%{filter_param}%'))
```

Comment:
+1, it's good to have a separate method for filtering. You are adding one more nested subquery but I would prefer to apply filters with same system query.
How about having a method that just build system related filters & apply those using filter method in main system_query?

Example - 

```
system_query = db.session.query(System.id).filter(
            System.account_id.in_(account_query)
        ).filter(*self.build_system_filters())
```

`build_system_filters` is static method which responsible to build system related filters as follows -
  ```
  @staticmethod
    def build_system_filters():
        """Build system filters."""
        filters = []
        if (filter_display_name := request.args.get('display_name')):
            filters.append(
                System.display_name.ilike(f'%{filter_display_name}%'))

        if (filter_by_state := request.args.get('state')):
            filters.append(System.state.ilike(f'%{filter_by_state}%'))
        else:
            filters.append(System.state.in_(SYSTEM_STATES_EXCEPT_EMPTY))

        return filters


```

@r14chandra - what do you think?  Will it be easy for one to add extra system filter using this method?

(Note - you can rename `build_system_filters` to something else I just provided randomly. :))

Reply:
@kgaikwad Yeah, we can do that. Adding this `build_system_filters` method in the code.